src/EnergyPlus/EcoRoofManager.cc

```diff
@@ -917,6 +919,34 @@ namespace EcoRoofManager {
             if (!state.dataGlobal->WarmupFlag) {
                 state.dataEcoRoofMgr->CumPrecip += state.dataEcoRoofMgr->CurrentPrecipitation;
             }
+        } else {
+            if (state.dataEnvrn->LiquidPrecipitation > 0.0) {
+                ShowWarningMessage(state,
+                                   "Please be aware that precipitation depth in the .epw weather file is used in the RoofIrrigation calculation as "
+                                   "the site:precipitation object is missing. Please make sure the precipitation depth in the weather file is valid. "
+                                   "Please refer to the 24-Hour Precipitation records by the State Climate Extremes Committee for a sanity check.");
+                state.dataEcoRoofMgr->CurrentPrecipitation = state.dataEnvrn->LiquidPrecipitation; //  units of m
+                // no site:precipitation, LiquidPrecipitation is zero but rain flag is on, assume 1.5mm rain
+            } else {
+                if (state.dataEnvrn->IsRain) {
+                    ShowWarningMessage(state, "Rain flag is on but precipitation in the weather file is missing, fill it with 1.5mm");
+                    state.dataEcoRoofMgr->CurrentPrecipitation = (1.5 / 1000.0); //  units of m
+                } else {
+                    state.dataEcoRoofMgr->CurrentPrecipitation = state.dataEnvrn->LiquidPrecipitation; //  units of m
+                }
+            }
```

Comment:
I expected to see these kinds of checks happening in the WeatherManager. My expectation was:
1. If there is a Site:Precipitation object present, use this to set `CurrentPrecipitation`
2. If not, then use the rainfall amounts from the epw file and use 1.5mm if rainflag = true and there is no precip data.

It appears that these warning could fire many times. If so, they should use the recurring warning method.

Reply:
Yeah it might be clearer to put the UpdatePrecipitation function in the weather manager instead. The checks will also be moved with UpdatePrecipitation.